rfc/2022/2022-08-23_CIT_hostnames.md

```diff
@@ -0,0 +1,227 @@
+
+All systems will need to be deployed and tested in Staging prior to a Production deployment.
+
+
+#### **Proposed Release Sequence:**
+
+**Step 1** (Rollback preparation)
+  - Identity Team works with IAM authentication partners to prepare a rollback plan for reverting auth callback changes. 
+  - Release Tools Team prepares a rollback PR and build.
+
+
+**Step 2** (Coordinated release of vets-api and vets-website repositories in lower environments)
+  - Broadcast an internal notification that Staging may be disrupted.
+  - Site Reliability Engineering notified of impending deployment to Staging.
+  - Infrastructure On-Call Team notified of impending deployment to Staging.
+  - In parallel:
+    - The Identity Team updates the vets-api authentication URLs for Dev and Staging. 
+    - The IAM team updates their own API callback URLs for Dev and Staging.
+  - Release Tools Team updates globally configured API URL for vets-website for Dev and Staging.
+  - vets-website repo:
+    - PRs merged into the main branch are automatically built and deployed to both dev and staging. 
+    - The GHA CI workflow runs. Completion time can vary, ~15 to 40 min.
+  - vets-api in Dev (EKS):
+    - Changes to the settings file are automatically deployed by ArgoCD. Completion time is fast.
+  - vets-api in Staging (BRD): 
+    - The Jenkins-led BRD cycle is triggered by PR merge to master, or manually if the change is initiated from the devops repo.
+    - The build step takes ~30 minutes, but can be run separately if needed.
+    - The deploy step is fast.
+  - Based on Pre-Release coordination research, trigger the deployment of both vets-api and vets-website so that they finish at approximately the same time. **Note**: Authentication flows may not work correctly until both applications have been fully deployed.
+
+
+
+**Step 3** (QA in Staging)
+  - Once the vets-api and vets-website Staging deployments are complete, Identity Team, Release Tools Team, and other affected front-end teams QA/UAT for issues. 
+  - Outstanding issues are fixed and re-deployed. Iterate as needed until Staging is stable.
+  - Once Staging is confirmed to be stable, prepare the Production deployment.
+
+
+**Step 4** (Coordinated release of vets-api and vets-website repositories in Production)
+  - Broadcast an internal notification that Production may be disrupted.
+  - Enable a sitewide, public-facing downtime notification informing users that Sign In will be unavailable.
+  - Site Reliability Engineering notified of impending deployment to Production.
+  - Infrastructure On-Call Team notified of impending deployment to Production.
+  - In parallel:
+    - The Identity Team prepares a PR for updates to the vets-api authentication URLs for Production.
+    - The IAM team updates the API callback URLs.
+  - Release Tools Team prepares a PR for updates to globally configured API URL for vets-website for Production.
```

Comment:
I think we need to note the references in vets-website that need to stay api.va.gov as well. There will need to be a lot of investigation here to figure out where the global references will break direct lighthouse calls

Reply:
@timwright12, other than in the `health-care-questionnaire` app, which is currently _not_ used in production, I don't think there are any direct Lighthouse calls in the code base. As several people have mentioned, vets-website is not set up as a Lighthouse client, so the only LH calls should be via vets-api. I don't know why that particular app started making those direct calls (and I also don't know if it ever worked) but perhaps @mdewey could shed some light on that.

Reply:
Yup that is my understanding as well, the reason being that Lighthouse requires usage of API keys therefore any queries going directly from FE -> Lighthouse would expose those API keys.